### PR TITLE
Added doc comments to System.Text.Encodings.Web

### PR DESCRIPTION
**Pull request by KrzysztofCwalina**
@stephentoub

### CODE REVIEW

File: src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/HtmlEncoder.cs [hunk 1]
```diff
@@ -10,16 +10,35 @@
 
 namespace System.Text.Encodings.Web
 {
+    /// <summary>
+    /// Represents a type used to do HTML encoding.
+    /// </summary>
     public abstract class HtmlEncoder : TextEncoder
     {
+        /// <summary>
+        /// Returns default built-in instance of HtmlEncoder.
```
Comment (stephentoub) [inline src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/HtmlEncoder.cs]: Nit: "a" default

File: src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/HtmlEncoder.cs [hunk 2]
```diff
@@ -10,16 +10,35 @@
...
+        /// </summary>
         public static HtmlEncoder Default
         {
             get { return DefaultHtmlEncoder.Singleton; }
         }
+
+        /// <summary>
+        /// Creates a new instalce of HtmlEncoder with provided settings.
```
Comment (stephentoub) [inline src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/HtmlEncoder.cs]: Nit: instalce => instance

Comment (stephentoub) [inline src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/HtmlEncoder.cs]: Nit: throughout you can also use XML like ```<see cref="HtmlEncode" />``` to enable tools to understand that you're referring to a class, provide appropriate hyperlinks, etc.

File: src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/HtmlEncoder.cs [hunk 3]
```diff
@@ -10,16 +10,35 @@
...
+        /// </summary>
+        /// <param name="settings">Settings used mainly to specify which characters end up being encoded.</param>
```
Comment (stephentoub) [inline src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/HtmlEncoder.cs]: Maybe ```Settings used to control how the created <see cref="HtmlEncode"/> encodes, primarily which characters to encode.``` or something like that.

File: src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/HtmlEncoder.cs [hunk 4]
```diff
@@ -10,16 +10,35 @@
...
+        /// <returns>Returns new instance of the encoder.</returns>
```
Comment (stephentoub) [inline src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/HtmlEncoder.cs]: I think you can leave off the word "Returns", e.g. ```A new instance of an <see cref="HtmlEncoder"/>.```

File: src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/HtmlEncoder.cs [hunk 5]
```diff
@@ -10,16 +10,35 @@
...
         public static HtmlEncoder Create(TextEncoderSettings settings)
         {
             return new DefaultHtmlEncoder(settings);
         }
+
+        /// <summary>
+        /// Creates a new instalce of HtmlEncoder specifying character to be encoded.
```
Comment (stephentoub) [inline src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/HtmlEncoder.cs]: Nit: search and replace "instalce" => "instance"

File: src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/HtmlEncoder.cs [hunk 6]
```diff
@@ -10,16 +10,35 @@
...
+        /// </summary>
+        /// <param name="allowedRanges">Set of characters that the endocer is allowed to not encode.</param>
```
Comment (stephentoub) [inline src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/HtmlEncoder.cs]: Search and replace "endocer" => "encoder"

File: src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/HtmlEncoder.cs [hunk 7]
```diff
@@ -10,16 +10,35 @@
...
+        /// <returns>Returns new instance of the encoder.</returns>
```
Comment (stephentoub) [inline src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/HtmlEncoder.cs]: Same comments as earlier.  I'll stop commenting on it.

File: src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/HtmlEncoder.cs [hunk 8]
```diff
@@ -10,16 +10,35 @@
...
+        /// <remarks>Some characters in the ranges might still get encoded, i.e. this parameter is just a hint.</remarks>
```
Comment (stephentoub) [inline src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/HtmlEncoder.cs]: This seems like strange behavior.  Could you elaborate with an example to help clarify? e.g. "This parameter is just a hint, and the encoder encode characters in these ranges.  For example, ...".

File: src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/TextEncoder.cs [hunk 9]
```diff
@@ -9,7 +9,14 @@
...
 
 namespace System.Text.Encodings.Web
-{    
+{
+    /// <summary>
+    /// An abstraction representing various text encoders. 
+    /// </summary>
+    /// <remarks>
+    /// TextEncoder subclasses can be used to do HTML encoding, URI encoding, and JavaScript encoding. 
+    /// Instances of such subclasses can be accessed using HtmlEncoder.Default, UrlEncoder.Default, and JavaScriptEncoder.Default.
```
Comment (stephentoub) [inline src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/TextEncoder.cs]: As with mentions of types, you can use ```<see cref="..."/>``` with members like Default.

File: src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/TextEncoder.cs [hunk 10]
```diff
@@ -19,6 +26,17 @@ public abstract class TextEncoder
         // yet consumption of concrete encoders is totally CLS compliant, 
         // as it’s mainly to be done by calling helper methods in TextEncoderExtensions class, 
         // and so I think the warning is a bit too aggressive.  
+        /// <summary>
```
Comment (stephentoub) [inline src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/TextEncoder.cs]: Nit: I'd suggest adding a blank line separating the normal comments from the XML comments

File: src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/TextEncoder.cs [hunk 11]
```diff
@@ -19,6 +26,17 @@ public abstract class TextEncoder
...
+        /// Encodes a unicode scalar into a buffer.
+        /// </summary>
+        /// <param name="unicodeScalar">Unicode scalar.</param>
+        /// <param name="buffer">The destination of the encoded text.</param>
+        /// <param name="bufferLength">Length of the destination buffer in chars.</param>
```
Comment (stephentoub) [inline src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/TextEncoder.cs]: You can use ```<paramref .../>``` to refer to other parameters.

File: src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/HtmlEncoder.cs (hunk 1 above)
Comment (KrzysztofCwalina) [inline src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/HtmlEncoder.cs]: fixed. thanks!

File: src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/HtmlEncoder.cs (hunk 2 above)
Comment (KrzysztofCwalina) [inline src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/HtmlEncoder.cs]: fixed. thanks!

File: src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/HtmlEncoder.cs (hunk 3 above)
Comment (KrzysztofCwalina) [inline src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/HtmlEncoder.cs]: fixed. thanks!

File: src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/HtmlEncoder.cs (hunk 4 above)
Comment (KrzysztofCwalina) [inline src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/HtmlEncoder.cs]: fixed. thanks!

File: src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/HtmlEncoder.cs (hunk 5 above)
Comment (KrzysztofCwalina) [inline src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/HtmlEncoder.cs]: fixed. thanks!

File: src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/HtmlEncoder.cs (hunk 6 above)
Comment (KrzysztofCwalina) [inline src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/HtmlEncoder.cs]: fixed. thanks!

File: src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/HtmlEncoder.cs (hunk 7 above)
Comment (KrzysztofCwalina) [inline src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/HtmlEncoder.cs]: Yeah, I fixed all instances of this.

File: src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/TextEncoder.cs (hunk 10 above)
Comment (KrzysztofCwalina) [inline src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/TextEncoder.cs]: done

File: src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/TextEncoder.cs (hunk 9 above)
Comment (KrzysztofCwalina) [inline src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/TextEncoder.cs]: changed.

File: src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/TextEncoder.cs (hunk 11 above)
Comment (KrzysztofCwalina) [inline src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/TextEncoder.cs]: changed.

File: src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/HtmlEncoder.cs (hunk 8 above)
Comment (KrzysztofCwalina) [inline src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/HtmlEncoder.cs]: This type is an abstraction, so it's not clear to me if examples would be useful. The point is that the allowedRanges is just telling the encoder what ranges it is allowed to not encode; it is not ordering it to not encode them. Implementations may still hardcode some special characters and to always encode them.

Maybe I just say: 
"Some characters in the ranges might still get encoded, i.e. this parameter is just telling the encoder what ranges it is allowed to not encode, not what characters it must not encode."

**Comment by KrzysztofCwalina** [comment]
@stephentoub, thanks for the review. I think I fixed all these now.

**Comment by KrzysztofCwalina** [comment]
@dotnet-bot test this please

**Pull request merged**